src/main/java/games/strategy/engine/framework/ui/GameChooserModel.java

```diff
@@ -172,23 +140,18 @@ private static void confirmWithUserAndThenDeleteCorruptZipFile(final File map, f
   /**
    * From a given URI, creates a GameChooserEntry and adds to the given entries list.
    *
-   * @param entries
-   *        list of entries where to add the new entry.
    * @param uri
    *        URI of the new entry
    */
-  private static void addNewGameChooserEntry(final List<GameChooserEntry> entries, final URI uri) {
+  private static Optional<GameChooserEntry> createGameChooserEntry(final URI uri) {
     try {
-      final GameChooserEntry newEntry = createEntry(uri);
-      if (!entries.contains(newEntry)) {
-        entries.add(newEntry);
-      }
+      return Optional.of(createEntry(uri));
     } catch (final EngineVersionException e) {
-      System.out.println(e.getMessage());
+      ClientLogger.logQuietly("Engine version problem:" + uri, e);
     } catch (final Exception e) {
-      System.err.println("Could not parse:" + uri);
```

Comment:
Note: Behavior change here.  A map parse error would have previously brought up the Error Console.  Now, unless Always Show Console is enabled, the user will not receive any notification.